File: Problem5_Tokyo/5_deepseek_sol1.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(System.out);

        int T = Integer.parseInt(br.readLine().trim());

        while (T-- > 0) {
            int N = Integer.parseInt(br.readLine().trim());

            long[] prefix = new long[N + 1];

            for (int i = 1; i <= N; i++) {
                String[] parts = br.readLine().trim().split("\\s+");
                char C = parts[0].charAt(0);
                long X = Long.parseLong(parts[1]);

                if (C == 'L') {
                    prefix[i] = prefix[i - 1] + X;
                } else {
                    prefix[i] = prefix[i - 1] - X;
                }
            }

            HashMap<Long, Integer> firstPos = new HashMap<>();
            firstPos.put(0L, 0);

            int bestLen = 0;
            int bestStart = 1;

            for (int i = 1; i <= N; i++) {
                long curr = prefix[i];

                if (firstPos.containsKey(curr)) {
                    int start = firstPos.get(curr) + 1;
                    int len = i - firstPos.get(curr);

                    if (len > bestLen || (len == bestLen && start < bestStart)) {
                        bestLen = len;
                        bestStart = start;
                    }
                } else {
                    firstPos.put(curr, i);
                }
            }

            if (bestLen == 0) {
                pw.println("0 0 0");
            } else {
                pw.println(bestLen + " " + bestStart + " " + (bestStart + bestLen - 1));
            }
        }

        pw.flush();
        pw.close();
        br.close();
    }
}

File: Problem5_Tokyo/5_deepseek_sol1.py
import sys


def solve():
    data = sys.stdin.read().strip().split()
    if not data:
        return

    idx = 0
    T = int(data[idx])
    idx += 1

    results = []

    for _ in range(T):
        N = int(data[idx])
        idx += 1

        prefix = [0] * (N + 1)

        for i in range(1, N + 1):
            C = data[idx]
            X = int(data[idx + 1])
            idx += 2

            if C == 'L':
                prefix[i] = prefix[i - 1] + X
            else:
                prefix[i] = prefix[i - 1] - X

        first_pos = {0: 0}
        best_len = 0
        best_start = 1

        for i in range(1, N + 1):
            curr = prefix[i]

            if curr in first_pos:
                start = first_pos[curr] + 1
                length = i - first_pos[curr]

                if length > best_len or (length == best_len and start < best_start):
                    best_len = length
                    best_start = start
            else:
                first_pos[curr] = i

        if best_len == 0:
            results.append("0 0 0")
        else:
            results.append(
                f"{best_len} {best_start} {best_start + best_len - 1}")

    sys.stdout.write("\n".join(results))


if __name__ == "__main__":
    solve()
